transaction/core/src/signer/traits.rs

```diff
@@ -0,0 +1,127 @@
+use crate::{
+    ring_signature::{CryptoRngCore, MLSAGError, ReducedTxOut, RingMLSAG, Scalar},
+    Amount,
+};
+use alloc::{string::String, vec::Vec};
+use displaydoc::Display;
+use mc_crypto_keys::{KeyError, RistrettoPrivate};
+use serde::{Deserialize, Serialize};
+use zeroize::Zeroize;
+
+/// A representation of the part of the input ring needed to create an MLSAG
+#[derive(Clone, Debug)]
+pub struct SignableInputRing {
+    /// A reduced representation of the TxOut's in the ring. For each ring
+    /// member we have only:
+    /// * The onetime-address (tx_out.target_key)
+    /// * The compressed commitment (tx_out.amount.commitment)
+    pub members: Vec<ReducedTxOut>,
+
+    /// The index of the real input among these ring members
+    pub real_input_index: usize,
+
+    /// The secrets needed to sign that input
+    pub input_secret: InputSecret,
+}
+
+/// The secrets needed to create a signature that spends an existing output as
+/// an input
+#[derive(Clone, Debug, Zeroize)]
+#[zeroize(drop)]
+pub struct InputSecret {
+    /// Represents either the one-time private key, or data to derive it
+    pub onetime_key_derive_data: OneTimeKeyDeriveData,
```

Comment:
Optional/Nit: It seems more consistent to name this `one_time_key_derive_data` (with `_` after `one`).

Reply:
yeah i agree based on the type name, but the thing is that over all in our code base we use `onetime_private_key` almost everywhere without an extra `_` after the one, so I feel like right now it's not more consistent. We could change the struct to `OnetimePrivateKey` I guess